staging/src/k8s.io/client-go/tools/clientcmd/validation.go

```diff
@@ -293,14 +311,55 @@ func validateAuthInfo(authInfoName string, authInfo clientcmdapi.AuthInfo) []err
 			} else {
 				defer clientCertFile.Close()
 			}
+			clientCertData, err := io.ReadAll(clientCertFile)
+			if err != nil {
+				validationErrors = append(validationErrors, fmt.Errorf("unable to read client-cert %v for %v due to %w", authInfo.ClientCertificate, authInfoName, err))
+			}
+			if len(clientCertData) == 0 {
+				validationErrors = append(validationErrors, fmt.Errorf("client-cert %v for %v is empty", authInfo.ClientCertificate, authInfoName))
+			} else {
+				if err := utilcert.ValidateCertificatePEM(clientCertData); err != nil {
```

Comment:
We can validate this contains a certificate block we can parse (if it doesn't cause issues by double-reading the file), but we shouldn't forbid files that currently work (ones that contain a cert and key block)